handshake_server.go

```diff
@@ -309,6 +311,11 @@ Curves:
 			hs.ecdsaOk = true
 		case *rsa.PublicKey:
 			hs.rsaSignOk = true
+		case ed25519.PublicKey:
+			if c.vers < VersionTLS13 {
+				c.sendAlert(alertInternalError)
+				return false, errors.New("tls: ed25519 only supported with version 1.3")
+			}
```

Comment:
As noted before, Ed25519 is being drafted for TLS 1.2 as well, so this check is probably wrong?

I think this should set `hs.ecdsaOk = true`. It is used to check whether a ciphersuite can be selected with the configured certificate. See https://tools.ietf.org/html/draft-ietf-tls-rfc4492bis-17#page-17

>   NOTE: SignatureAlgorithm is "rsa" for the ECDHE_RSA key exchange
>   algorithm and "anonymous" for ECDH_anon.  These cases are defined in
>   TLS.  SignatureAlgorithm is "ecdsa" or "eddsa" for ECDHE_ECDSA.